internal/errors/resolver/resolver.go

```diff
@@ -28,9 +28,14 @@ func AddErrorResolver(er ErrorResolver) {
 // the error could not be resolved.
 func ResolveError(err error) (ResolvedResult, bool) {
 	for _, resolver := range errorResolvers {
-		msg, found := resolver.Resolve(err)
+		rr, found := resolver.Resolve(err)
+		// If the exit code hasn't been set, we default it to 1. We should
+		// never return exit code 0 for errors.
+		if rr.ExitCode == 0 {
```

Comment:
Changing `ExitCode` to pointer would have been probably better. But I can't think of a use-case where we will return zero, so it's probably fine.

Reply:
I considered that, but pointers to scalars are awkward.